packages/firebase/externs/firebase-auth-externs.js

```diff
@@ -1382,7 +1434,46 @@ firebase.auth.Auth.prototype.signInAndRetrieveDataWithCredential = function(
 ) {};
 
 /**
- * Asynchronously signs in using a custom token.
+ * Signs in a user asynchronously using a custom token and returns any
+ * additional user info data or credentials.
+ *
+ * Custom tokens are used to integrate Firebase Auth with existing auth systems,
+ * and must be generated by the auth backend.
+ *
+ * Fails with an error if the token is invalid, expired, or not accepted by the
+ * Firebase Auth service.
+ *
+ * <h4>Error Codes</h4>
+ * <dl>
+ * <dt>auth/custom-token-mismatch</dt>
+ * <dd>Thrown if the custom token is for a different Firebase App.</dd>
+ * <dt>auth/invalid-custom-token</dt>
+ * <dd>Thrown if the custom token format is incorrect.</dd>
+ * </dl>
+ *
+ * @example
+ * firebase.auth().signInAndRetrieveDataWithCustomToken(token)
+ *     .catch(function(error) {
+ *       // Handle Errors here.
+ *       var errorCode = error.code;
+ *       var errorMessage = error.message;
+ *       if (errorCode === 'auth/invalid-custom-token') {
+ *         alert('The token you provided is not valid.');
+ *       } else {
+ *         console.error(error);
+ *       }
+ *     });
+ *
+ * @param {string} token The custom token to sign in with.
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAndRetrieveDataWithCustomToken = function(
+  token
+) {};
+
+/**
+ * Asynchronously signs in using a custom token. Will be depercated, replaced by
```

Comment:
This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAndRetrieveDataWithCustomToken}.